candle-transformers/src/models/t5.rs

```diff
@@ -348,9 +348,14 @@ impl T5Attention {
                 None => (scores, None),
                 Some(relative_attention_bias) => {
                     // This only handles the bidirectional case.
+                    let kv_len = k.dim(2)?;
```

Comment:
This doesn't seem right to me for the case where `use_cache=true` and the query size is larger than 1 (typically the case when processing a prompt). I would think it's more something like `kv_len - q_len` to `kv_len`?

Reply:
Sorry, I'm not sure if I follow.

I added a new `decoder_prompt` to get to this scenario where `use_cache=true` and query size is larger than 1 in both the encoder and decoder and it works fine:

```bash
$ cargo run --release --example t5 -- --model-id "t5-base" --prompt "summarize: state authorities dispatched emergency crews tuesday to survey the damage after an onslaught of severe weather in mississipi." --decoder-prompt "the weather has" --temperature 0 --decode
    Finished release [optimized] target(s) in 0.17s
     Running `target/release/examples/t5 --model-id t5-base --prompt 'summarize: state authorities dispatched emergency crews tuesday to survey the damage after an onslaught of severe weather in mississipi.' --decoder-prompt 'the weather has' --temperature 0 --decode`
Running on CPU, to run on GPU, build this example with `--features cuda`
the weather has been severe in mississipi . state authorities dispatched emergency crews to survey the damage .
29 tokens generated (15.39 token/s)
```

Reply:
Isn't the `q_start=kv_len-1`, and `q_end=kv_len` assuming a query length of 1?

Reply:
(ps: this code is never executed for cross-attention, so the variable names may be a bit misleading)

Reply:
Oh, I see what you mean, sorry. I was looking at the line 351, not 352.

Not sure how it was working before 🤔 . Fixed it now.